### PR TITLE
Collision estimator

### PR DESCRIPTION
**Pull request by smharper**
This PR gives us collision estimators for some scores.  I haven't done anything fancy like sampling outgoing energies so the collision estimator is only available in the same cases that tracklength estimator is available.

I've added support for the collision estimator in the docs, PythonAPI, and test suite as well as the Fortran source.  I've also added a bunch of analog estimators to the test suite.

What do you guys want to do with the analog flux estimator?  The analog flux estimator is really a collision flux estimator, but I'm not sure if we want to remove it because then flux will be the only score that you can't use an analog estimator with.

I've verified that all three estimators converge to the same results for a pincell calculation.  I tried making a pretty graphic to show this, but I couldn't get a graphic I was happy with so here is just a raw dump of my tally results:
```
 =====================>     TALLY 1: TRACKLENGTH     <======================

 Incoming Energy [0.0, 4.00000E-06)
   Total Material
     Flux                                 25.0750        +/- 8.11266E-03
     Fission Rate                         5.89867E-02    +/- 6.82288E-05
     Nu-Fission Rate                      0.143733       +/- 1.66253E-04
     Scattering Rate                      53.8984        +/- 1.88569E-02
     Absorption Rate                      0.968364       +/- 3.49177E-04
 Incoming Energy [4.00000E-06, 20.0000)
   Total Material
     Flux                                 26.4465        +/- 5.07465E-03
     Fission Rate                         8.49039E-03    +/- 5.88953E-06
     Nu-Fission Rate                      2.28831E-02    +/- 1.59039E-05
     Scattering Rate                      15.7688        +/- 2.17830E-03
     Absorption Rate                      3.23966E-02    +/- 3.43816E-05

 ========================>     TALLY 2: ANALOG     <========================

 Incoming Energy [0.0, 4.00000E-06)
   Total Material
     Flux                                 25.0778        +/- 8.55105E-03
     Fission Rate                         5.90299E-02    +/- 7.67783E-05
     Nu-Fission Rate                      0.143824       +/- 1.86132E-04
     Scattering Rate                      53.9002        +/- 1.90183E-02
     Absorption Rate                      0.967956       +/- 6.35764E-05
 Incoming Energy [4.00000E-06, 20.0000)
   Total Material
     Flux                                 26.4460        +/- 5.69557E-03
     Fission Rate                         8.47943E-03    +/- 3.05203E-05
     Nu-Fission Rate                      2.29177E-02    +/- 3.79627E-05
     Scattering Rate                      15.7700        +/- 1.69404E-03
     Absorption Rate                      3.23332E-02    +/- 6.32495E-05

 ======================>     TALLY 3: COLLISION     <=======================

 Incoming Energy [0.0, 4.00000E-06)
   Total Material
     Flux                                 25.0778        +/- 8.55105E-03
     Fission Rate                         5.89774E-02    +/- 6.86483E-05
     Nu-Fission Rate                      0.143710       +/- 1.67275E-04
     Scattering Rate                      53.8998        +/- 1.87018E-02
     Absorption Rate                      0.968365       +/- 3.40849E-04
 Incoming Energy [4.00000E-06, 20.0000)
   Total Material
     Flux                                 26.4460        +/- 5.69557E-03
     Fission Rate                         8.49170E-03    +/- 8.94506E-06
     Nu-Fission Rate                      2.28888E-02    +/- 2.43755E-05
     Scattering Rate                      15.7700        +/- 1.67496E-03
     Absorption Rate                      3.23333E-02    +/- 3.23313E-05
```

### CODE REVIEW

**Comment by smharper** [comment]
Added blurb in the docs explaining the analog-collision flux score thing.  This also seems to have kicked Travis into action.  Not sure why it wasn't testing before that commit

File: src/input_xml.F90 [hunk 1]
```diff
@@ -3138,6 +3138,18 @@ subroutine read_tallies_xml()
...
+          ! If the estimator was set to an analog estimator, this means the
+          ! tally needs post-collision information
+          if (t % estimator == ESTIMATOR_ANALOG) then
+            call fatal_error("Cannot use collision estimator for tally " &
+                 &// to_str(t % id))
+          end if
+
+          ! Set estimator to collision estimator
+          t % estimator = ESTIMATOR_COLLISION
+          write(*, *) t % estimator
```
Comment (paulromano) [inline src/input_xml.F90]: Remove write statement

File: src/input_xml.F90 [hunk 2]
```diff
@@ -3138,6 +3138,18 @@ subroutine read_tallies_xml()
           ! Set estimator to track-length estimator
           t % estimator = ESTIMATOR_TRACKLENGTH
 
+        case ('collision')
+          ! If the estimator was set to an analog estimator, this means the
+          ! tally needs post-collision information
+          if (t % estimator == ESTIMATOR_ANALOG) then
+            call fatal_error("Cannot use collision estimator for tally " &
+                 &// to_str(t % id))
```
Comment (paulromano) [inline src/input_xml.F90]: ampersand starting continuation line is unnecessary

File: src/tally.F90 [hunk 3]
```diff
@@ -92,7 +92,8 @@ subroutine score_general(p, t, start_index, filter_index, i_nuclide, &
           end if
           score = score / material_xs % total
 
-        else if (t % estimator == ESTIMATOR_TRACKLENGTH) then
+        else if (t % estimator == ESTIMATOR_TRACKLENGTH .or. &
+                 t % estimator == ESTIMATOR_COLLISION) then
```
Comment (paulromano) [inline src/tally.F90]: All of these "else if tracklength or collision" could probably just be "else". No comparisons needed, although it is at the expense of clarity.

**Comment by paulromano** [comment]
I'd say just leave the analog flux estimator alone. Like you said, it's a little weird if that's the only estimator that can't do flux. We could print out a warning message if the user asks for an analog flux and just tell them that they're actually getting a collision estimate.

Are you planning on implementing the ability to sample angles/energies for collision/tracklength tallies so they can be used for all tallies? If not, let's add an issue for that.

I don't see any other obvious problems with your code, so I think it's ready to go once those few minor issues are resolved. Thanks!

**Comment by smharper** [comment]
I've addressed your line notes in that latest commit.

Re warning message: I think that such a warning message would mostly be annoying clutter.  I think the note in the docs should be enough.

Re sampling: Not planning on it.  I'll open an issue shortly.

**Pull request merged**